crates/wasm-mutate/src/lib.rs

```diff
@@ -246,37 +253,45 @@ impl WasmMutate {
                     let mut func = Vec::new();
 
                     WasmMutate::mutate(
-                        &mut payload, &mut rnd, &self, byteschunk, &mut func);
-
-                    code.function_raw(&func);
+                        &mut payload, self, byteschunk, &mut func);
 
-        
+                    code.extend(&func);
                 },
                 _ => {
                     WasmMutate::mutate(
-                        &mut payload, &mut rnd, &self, byteschunk,     
+                        &mut payload, &self, byteschunk,     
                         if !code_parsing_started {
                             &mut first_half
                         } else {
                             &mut second_half
                         });
                 }
             }
-
-
             consumed += chunksize
         }
 
         // Write all to result buffer before the code section started
-        result.write(first_half);
+        result.extend(first_half);
         // Recreate the code section
         if code_parsing_started  {
-            let mut module = Module::new();
-            module.section(&code);
-            result.write(&module.finish()).expect("Code section could not be written");
+            // Add function count and section size to raw data
+            let mut sink = Vec::new();
+            let num_added = encoders::u32(function_count);
+            sink.extend(num_added);
+            sink.extend(code.iter().copied());
+            
+            let raw_code = RawSection{
+                id: SectionId::Code.into(),
+                data: &sink
+            };
+
+            // To check, add section if in the encoding of the section ?
+            result.extend_one(SectionId::Code as u8);
```

Comment:
You can do `.extend(std::iter::once(..))`

Reply:
Solved